workbench-core/environments-ui/src/pages/environments/new.tsx

```diff
@@ -0,0 +1,337 @@
+import EnvTypeCards from '../../components/EnvTypeCards';
+import EnvTypeConfigCards from '../../components/EnvTypeConfigCards';
+import { CreateEnvironmentForm, CreateEnvironmentFormValidation } from '../../models/Environment';
+import { EnvTypeItem } from '../../models/EnvironmentType';
+import { EnvTypeConfigItem } from '../../models/EnvironmentTypeConfig';
+
+const NewEnvironmentPage: NextPage = () => {
+  // App settings constant
+
+  const router = useRouter();
+  const [isSubmitLoading, setIsSubmitLoading] = useState(false);
+  const [disableSubmit, setDisableSubmit] = useState(true);
+  const [selectedEnvType, setSelectedEnvType] = useState<EnvTypeItem>();
+  const [error, setError] = useState('');
+  const [formData, setFormData] = useState<CreateEnvironmentForm>({});
+  const [formErrors, setFormErrors] = useState<CreateEnvironmentFormValidation>({});
+  const { envTypes, areEnvTypesLoading } = useEnvironmentType();
+  const { envTypeConfigs, areEnvTypeConfigsLoading } = useEnvTypeConfigs(formData?.envTypeId || '');
+  const { projects, areProjectsLoading } = useProjects();
+  const { datasets, areDatasetsLoading } = useDatasets();
+
+  const breadcrumbs: BreadcrumbGroupProps.Item[] = [
+    {
+      text: 'Service Workbench',
+      href: '/'
+    },
+    {
+      text: 'Workspaces',
+      href: '/environments'
+    },
+    {
+      text: 'Create Workspace',
+      href: '/environments/new'
+    }
+  ];
+  const validationRules = [
+    {
+      field: 'name',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a,
+      message: 'Workspace Name is Required'
+    },
+    {
+      field: 'name',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a && nameRegex.test(a),
+      message:
+        'Workspace Name must start with an alphabetic character and can only contain alphanumeric characters (case sensitive) and hyphens.'
+    },
+    {
+      field: 'name',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a && a.length <= 128,
+      message: 'Workspace Name cannot be longer than 128 characters'
+    },
+    {
+      field: 'projectId',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a,
+      message: 'Project ID is Required'
+    },
+    {
+      field: 'envTypeId',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a,
+      message: 'Compute Platform is Required'
+    },
+    {
+      field: 'envTypeConfigId',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !!a,
+      message: 'Configuration is Required'
+    },
+    {
+      field: 'description',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      condition: (a: any) => !a || a.length <= 500,
```

Comment:
Fun fact: I _think_ if you did this as an `&&` and `a` was false, the interpreter would stop checking the rest of the argument after the first false condition, allegedly saving you some time. That is obviously an extreme over optimization for this situation, but  I just always thought that was neat optimization.